explorer/mempool.go

```diff
@@ -147,11 +172,31 @@ func (exp *explorerUI) storeMempoolInfo() {
 	var totalOut float64
 	var totalSize int32
 
+	exp.NewBlockDataMtx.Lock()
```

Comment:
Consider whether a full read-write lock (as with `Lock()`) is needed or if a read lock is good here (`RLock()`).  And please move the unlock as close as possible to the lock.  That is, can it go before the call to `GetBlockHash`, which may have slow IO.